src/grpc/stream_infer_handler.cc

```diff
@@ -331,6 +348,8 @@ ModelStreamInferHandler::Process(InferHandler::State* state, bool rpc_ok)
         state->complete_ = true;
         state->context_->PutTaskBackToQueue(state);
       }
+    } else {
+      state->context_->InsertInflightState(state);
```

Comment:
Can you swap this so the first condition `if err == nullptr`, and the error handling is the `else` ? Also, split the comment across each condition:
```cpp
if (err == nullptr) {
    // If there was not an error in issuing the 'state' request then
    // state->step_ == ISSUED and inference request has
    // initiated, so mark this state as inflight.
    // The completion callback will transition to
    // WRITEREADY or WRITTEN.
    state->context_->InsertInflightState(state);
} else {
    // If there was an error then enqueue the
    // error response and show it to be ready for writing.
    ...
}
```